src/supremm/datasource/pcp/pcpdatasource.py

```diff
@@ -0,0 +1,101 @@
+import os
+import shutil
+import time
+import logging
+import datetime
+
+from supremm.datasource.datasource import Datasource
+from supremm.datasource.pcp.pcparchive import extract_and_merge_logs
+from supremm.datasource.pcp.pcpsummarize import PCPSummarize
+from supremm.errors import ProcessingError
+
+PCP_STR = "pcp"
+
+
+class PCPDatasource(Datasource):
+    """ Instance of a PCP datasource class """
+
+    def __init__(self, preprocs, plugins):
+        super().__init__(preprocs, plugins)
+
+    def presummarize(self, job, conf, resconf, opts):
+        jobmeta = super().presummarize(job, conf, resconf, opts)
+
+        # Error with general presummarize, don't try datasource specific checks
+        if jobmeta.result != 0 and jobmeta.error != None:
+            return jobmeta
+        else:
+            mergestart = time.time()
+            if not job.has_any_archives():
+                jobmeta.result = 1
+                jobmeta.mdata["skipped_noarchives"] = True
+                jobmeta.error = ProcessingError.NO_ARCHIVES
+                jobmeta.missingnodes = job.nodecount
+                logging.info("Skipping %s, skipped_noarchives", job.job_id)
+            elif not job.has_enough_raw_archives():
+                jobmeta.result = 1
+                jobmeta.mdata["skipped_rawarchives"] = True
+                jobmeta.error = ProcessingError.RAW_ARCHIVES
+                jobmeta.missingnodes = job.nodecount
+                logging.info("Skipping %s, skipped_rawarchives", job.job_id)
+            else:
+                jobmeta.result = extract_and_merge_logs(job, conf, resconf, opts)
+                jobmeta.missingnodes = -1.0 * jobmeta.result
+
+        mergeend = time.time()
+        jobmeta.mdata["mergetime"] = mergeend - mergestart
+
+        if opts['extractonly']:
+            if jobmeta.result == 0:
+                return None
+            else:
+                logging.error("Failure extracting logs for job %s", job.job_id)
+                return None
+
+        return jobmeta
+
+    def summarizejob(self, job, jobmeta, conf, opts):
+        preprocessors, analytics = super().summarizejob(job, jobmeta, conf, opts)
+
+        s = PCPSummarize(preprocessors, analytics, job, conf, opts["fail_fast"], PCP_STR)
```

Comment:
The prototype for PCPSumarize is this:
```
def __init__(self, preprocessors, analytics, job, config, datasource, fail_fast=False):
```

the function arguments don't match what is being called here: PCP_STR and fail_fast have been transposed.

My question is why do you pass in PCP_STR to PCPSummarize? Is there ever a scenario where the pcp-specific summarization code will use a different datasource string than PCP_STR?